### PR TITLE
[1.15] Vertical View Distance

### PR DESCRIPTION
**Pull request by AidanLovelace**
This new PR was created because I did some git nonos on the other branch so I recreated the changes on this branch.

Issue: #519

### CODE REVIEW

**Comment by NotStirred** [comment]
This is not working correctly, it seems to continuously load cubes around the player. Even at render distance 2h2v the same thing happens

I think it was working on b298f88

**Comment by AidanLovelace** [comment]
I think I got it properly working now. There were even more oversights. Guess I needed a week not looking at code lol.

File: src/main/java/io/github/opencubicchunks/cubicchunks/CubicChunksConfig.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+public class CubicChunksConfig {
+    private static final ForgeConfigSpec.Builder builder = new ForgeConfigSpec.Builder();
+    public static final ForgeConfigSpec config;
+
+    public static ForgeConfigSpec.IntValue verticalViewDistance;
+
+    static {
+        verticalViewDistance = builder
+                .comment("Vertical radius of cubes to load")
+                .defineInRange("verticalViewDistance", 4, 2, 16);
```
Comment (NotStirred) [inline src/main/java/io/github/opencubicchunks/cubicchunks/CubicChunksConfig.java]: Match vanilla view distance range
```suggestion
                .defineInRange("verticalViewDistance", 4, 2, 32);
```

File: src/main/java/io/github/opencubicchunks/cubicchunks/chunk/ticket/PlayerCubeTicketTracker.java [hunk 2]
```diff
@@ -11,81 +12,127 @@
...
+                int oldVertDistance = this.vertDistances.get(pos);
+                int horizCurrentDistance = horizontalGraphGroup.getLevel(pos);
+                int vertCurrentDistance = verticalGraphGroup.getLevel(pos);
+
+                if (oldHorizDistance != horizCurrentDistance || oldVertDistance != vertCurrentDistance) {
                     //func_219066_a = update level
-                    iTicketManager.getCubeTaskPriorityQueueSorter().onUpdateCubeLevel(CubePos.from(i), () -> this.distances.get(i), k, (ix) -> {
-                        if (ix >= this.distances.defaultReturnValue()) {
-                            this.distances.remove(i);
+                    iTicketManager.getCubeTaskPriorityQueueSorter().onUpdateCubeLevel(CubePos.from(pos), () -> this.horizDistances.get(pos), horizCurrentDistance, (horizDistance) -> {
```
Comment (NotStirred) [inline src/main/java/io/github/opencubicchunks/cubicchunks/chunk/ticket/PlayerCubeTicketTracker.java]: Take vertical view distance into account when sorting cube load tasks
```suggestion
                    iTicketManager.getCubeTaskPriorityQueueSorter().onUpdateCubeLevel(CubePos.from(pos), () -> this.horizDistances.get(pos), horizCurrentDistance+verticalViewDistance, (horizDistance) -> {
```

**Review by NotStirred — CHANGES_REQUESTED** [review]
I assume you've rebased to most recent commit, so I haven't addressed those changes.

With a view distance of 8h8v the vertical view is 2 larger than horiz as shown by the vis, so something is not quite right there

File: build.gradle [hunk 3]
```diff
@@ -132,7 +132,7 @@ repositories{
 configurations.getByName("annotationProcessor").extendsFrom(configurations.compileClasspath)
 
 // for vulkan
-project.ext.lwjglVersion = "3.2.2"
+project.ext.lwjglVersion = "3.2.1"
```
Comment (NotStirred) [inline build.gradle:135]:
```suggestion
project.ext.lwjglVersion = "3.2.2"
```

File: src/main/java/io/github/opencubicchunks/cubicchunks/mixin/core/client/MixinVideoSettingsScreen.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Redirect;
+
+@Mixin(VideoSettingsScreen.class)
+public class MixinVideoSettingsScreen {
+
+    private static final AbstractOption[] MODIFIED_OPTIONS;
+
+    private static final SliderPercentageOption VERTICAL_RENDER_DISTANCE = new SliderPercentageOption("options.renderDistance", 2.0D, 16.0D, 1.0F, (gameSettings) -> (double) CubicChunksConfig.verticalViewDistance.get(),
```
Comment (NotStirred) [inline src/main/java/io/github/opencubicchunks/cubicchunks/mixin/core/client/MixinVideoSettingsScreen.java:18]:
```suggestion
    private static final SliderPercentageOption VERTICAL_RENDER_DISTANCE = new SliderPercentageOption("options.renderDistance", 2.0D, 32.0D, 1.0F, (gameSettings) -> (double) CubicChunksConfig.verticalViewDistance.get(),
```

**Review by NotStirred — COMMENTED** [review]
Having issues with changing view distance not updating.
Also vertical view of 2 seems to break everything.

**Comment by Niko-sk2x** [comment]
Merged manually, finished and fixed in https://github.com/OpenCubicChunks/CubicChunks/commit/d41a712714c19754ed0d88ded0cc0550dae2c9a7

**Pull request closed without merging**